### PR TITLE
#283: Implemented password verification for login call

### PR DESCRIPTION
**Pull request by ekondrashev**
#283: Implemented password verification for login call

Enhanced test to check success login

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#419](https://github.com/timelesslounge/timelessis/issues/419) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#419](https://github.com/timelesslounge/timelessis/issues/419) is assigned to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/c44312c4-1291-48fc-899f-8f4edcc8af78); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_auth_test.py [hunk 1]
```diff
@@ -28,25 +30,22 @@ def test_incorrect_password(db_session):
...
     db_session.commit()
-    """error = login("vgv", "pass")"""
-    error = login("unknown", "unknown");
+    error = login("vgv", "pass")
+    assert not error
+    assert flask.session['user_id'] == employee.id
     db_session.delete(employee)
     db_session.commit()
     db_session.remove()
+    error = login("unknown", "unknown");
```
Comment (ammaratef45) [inline tests/integration/it_auth_test.py]: @ekondrashev do you need to remove the employee before testing `unknown` login?

**Comment by ammaratef45** [comment]
@ekondrashev great, had only one comment, check it.

File: tests/integration/it_auth_test.py (hunk 1 above)
Comment (ekondrashev) [reply tests/integration/it_auth_test.py]: @ammaratef45 that is what test was doing, I believe it is a part of a cleanup, so I'd move this to try/finally, but yes, the deletion is necessary imo

Comment (ammaratef45) [reply tests/integration/it_auth_test.py]: @ekondrashev there is a fixture that empty the DB I think, need to revise that
But at least make it in a final block or a fixture like you said

Comment (ammaratef45) [inline tests/integration/it_auth_test.py]: @ekondrashev Why there is a semicolon here? I know it's here before your edit, we should remove it, right?

**Comment by ammaratef45** [comment]
@rultor merge

**Comment by ammaratef45** [comment]
@ekondrashev thanks

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor stop

**Comment by emilianodellacasa** [comment]
@ekondrashev Please add a puzzle to the login page to add style to get following UX:

https://drive.google.com/uc?export=view&id=1oh8cLv-vHTCBmb0RDkHQiyMythC4_M6o

**Comment by ammaratef45** [comment]
@emilianodellacasa why don't you open an issue?

**Comment by emilianodellacasa** [comment]
@ammaratef45 Because this way PDD will recursevly group all related issues and it will be easier to check advancement. Also, if I open an issue (and I could) I would get bonus for issue creating - I am kinda altruistic and prefer to spare budget for you guys :-)

**Comment by ammaratef45** [comment]
@ekondrashev I think PDD is complaining about the puzzle

**Comment by ekondrashev** [comment]
@ammaratef45 @emilianodellacasa added a puzzle, please see if all is fine now

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#419](https://github.com/timelesslounge/timelessis/issues/419) is not in the agenda of @ammaratef45[/z](https://www.0crat.com/u/ammaratef45), can't inspect

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @ammaratef45[/z](https://www.0crat.com/u/ammaratef45), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#419](https://github.com/timelesslounge/timelessis/issues/419) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ammaratef45** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/419#issuecomment-468920556))

@ammaratef45 This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#419](https://github.com/timelesslounge/timelessis/issues/419) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 1hr
  * @ammaratef45[/z](https://www.0crat.com/u/ammaratef45) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23419%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@ammaratef45 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ammaratef45** [comment]
> @ammaratef45 According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
> 
> > The code reviewer found at least three problems in the code.
> > Comments were mostly about design problems, not cosmetic issues.
> 
> Only two issues were found during code review.
> Please confirm that you'll try to find at least three major problems while future reviews.

@ypshenychka Isn't two better than one? 😆 
I confirm

**Comment by ypshenychka** [comment]
@ammaratef45 of course better :) thanks!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Pull request merged**